### ISSUE TITLE
Reword documentation for Security/CompoundHash

### ISSUE THREAD

**Issue by gettalong** [rubocop/rubocop#11357]
The documentation for the cop [`Security/CompoundHash`](https://docs.rubocop.org/rubocop/1.41/cops_security.html#securitycompoundhash) has the following sentence in it:

> Delegating to Array#hash is clearer, faster, and safer.

This sentence is factually false because using `Array#hash` is not (always) faster, and benchmarking the given examples for good or bad even shows it:

~~~ yaml
prelude: |
  oid = 15
  gen = 0

benchmark:
  'with array': '[oid, gen].hash'
  'xor': 'oid ^ gen'
~~~

Running the benchmark file with `benchmark-driver` yields the following result;

~~~
Warming up --------------------------------------
          with array     8.800M i/s -      8.953M times in 1.017387s (113.63ns/i, 289clocks/i)
                 xor    27.050M i/s -     27.215M times in 1.006113s (36.97ns/i, 44clocks/i)
Calculating -------------------------------------
          with array    11.292M i/s -     26.401M times in 2.338090s (88.56ns/i, 248clocks/i)
                 xor    43.866M i/s -     81.149M times in 1.849915s (22.80ns/i, 26clocks/i)

Comparison:
                 xor:  43866097.2 i/s
          with array:  11291728.7 i/s - 3.88x  slower
~~~

I would reword the sentence similar to the following:

> Delegating to Array#hash is clearer and safer, although it might be slower depending on the use case.

**Comment by koic** [comment]
Definitely! I've confirm that and updated the doc by https://github.com/rubocop/rubocop/commit/aa03bb9883ba5ad034eb8930ab643afd0df05584. Thank you for the feedback!
